src/jshint.js

```diff
@@ -1419,12 +1425,23 @@ var JSHINT = (function() {
     var x = symbol(s, 150);
 
     x.led = function(left) {
+      // this = suffix e.g. "++" punctuator
+      // left = symbol operated e.g. "a" identifier or "a.b" punctuator
+
       if (state.option.plusplus) {
         warning("W016", this, this.id);
       } else if ((!left.identifier || isReserved(left)) && left.id !== "." && left.id !== "[") {
         warning("W017", this);
       }
 
+      // detect modification of a const.
+      // in the case of a.b, left will be the "." punctuator
```

Comment:
> It’s called for a number of prefixes, preinc, predec, unary minus, unary plus. The unary ops are totally fine because they’re not assignment ops.

you put this comment on the postfix function, which is only called for ++ and --

The prefix function this is the case, but in that function my block is clearly in the `if(this.id === "--" || this.id === "++")` section. anyway, as I said I've changed the comment anyway in both places, is that what you want?